operate/schema/src/main/java/io/camunda/operate/schema/IndexSchemaValidator.java

```diff
@@ -135,4 +147,92 @@ public boolean schemaExists() {
+        IndexMappingDifference difference = null;
+        // compare every index in group
+        for (Map.Entry<String, IndexMapping> singleIndexMapping : indexMappingsGroup.entrySet()) {
+          final IndexMappingDifference currentDifference =
+              indexMappingMustBe.difference(singleIndexMapping.getValue());
+          if (!currentDifference.isEqual()) {
+            if (difference == null) {
+              difference = currentDifference;
+            } else if (!difference.equals(currentDifference)) {
+              throw new OperateRuntimeException(
+                  "Ambiguous schema update. First bring runtime and date indices to one schema. Difference 1: "
+                      + difference
+                      + ". Difference 2: "
+                      + currentDifference);
+            }
+          }
+        }
+        if (difference != null && !difference.isEqual()) {
+          logger.debug(
+              String.format(
+                  "Index fields differ from expected. Index name: %s. Difference: %s.",
+                  indexDescriptor.getIndexName(), difference));
+          if (!difference.getEntriesDiffering().isEmpty()) {
+            final String errorMsg =
+                String.format(
+                    "Index name: %s. Not supported index changes are introduced. Data migration is required. Changes found: %s",
+                    indexDescriptor.getIndexName(), difference.getEntriesDiffering());
+            logger.error(errorMsg);
+            throw new OperateRuntimeException(errorMsg);
+          } else if (!difference.getEntriesOnlyOnRight().isEmpty()) {
+            final String message =
+                String.format(
+                    "Index name: %s. Field deletion is requested, will be ignored. Fields: %s",
+                    indexDescriptor.getIndexName(), difference.getEntriesOnlyOnRight());
+            logger.warn(message);
+          } else if (!difference.getEntriesOnlyOnLeft().isEmpty()) {
+            newFields.put(indexDescriptor, difference.getEntriesOnlyOnLeft());
+          }
+        } else {
+          logger.debug(
+              String.format(
+                  "Index fields are up to date. Index name: %s.", indexDescriptor.getIndexName()));
+        }
+      }
+    }
+    return newFields;
+  }
+
+  /**
+   * Leave only runtime and dated indices that correspond to the given IndexDescriptor.
+   *
+   * @param indexMappings
+   * @param indexDescriptor
+   * @return
+   */
+  private Map<String, IndexMapping> filterIndexMappings(
+      final Map<String, IndexMapping> indexMappings, final IndexDescriptor indexDescriptor) {
+    return Maps.filterEntries(
+        indexMappings,
+        e ->
+            e.getKey()
+                .matches(
+                    DatabaseInfo.getIndexPrefix(operateProperties)
+                        + "-"
+                        + indexDescriptor.getIndexName()
+                        + "-\\d.*"));
```

Comment:
Should we move this to `IndexDescriptor` or its subclasses so that all the naming patterns are in one place?